### PR TITLE
rustfmt for libgraphviz

### PR DESCRIPTION
**Pull request by scanfield**
Run rustfmt on libgraphviz

Tested: make -j8 check-stage1-std

Also ran rustfmt a second time to verify the transformation was stable.

### CODE REVIEW

**Comment by stevencanfield** [comment]
r? @nrc

Not sure about the raw strings around line 1037 in the lhs -- do you have advice for that?

File: src/libgraphviz/lib.rs [hunk 1]
```diff
@@ -63,7 +63,8 @@
 //! }
 //!
 //! impl<'a> dot::Labeller<'a, Nd, Ed> for Edges {
-//!     fn graph_id(&'a self) -> dot::Id<'a> { dot::Id::new("example1").unwrap() }
+//! fn graph_id(&'a self) -> dot::Id<'a> {
```
Comment (tamird) [inline src/libgraphviz/lib.rs]: this doesn't look right either

Comment (nrc) [inline src/libgraphviz/lib.rs]: https://github.com/nrc/rustfmt/issues/554

File: src/libgraphviz/lib.rs [hunk 2]
```diff
@@ -916,7 +924,7 @@ mod tests {
         let labels: Trivial = UnlabelledNodes(0);
         let r = test_input(LabelledGraph::new("empty_graph", labels, vec![], None));
         assert_eq!(r.unwrap(),
-r#"digraph empty_graph {
+                   r#"digraph empty_graph {
```
Comment (nrc) [inline src/libgraphviz/lib.rs]: Here and below: https://github.com/nrc/rustfmt/issues/559

File: src/libgraphviz/lib.rs [hunk 3]
```diff
@@ -1034,17 +1042,11 @@ r#"digraph hasse_diagram {
 
     #[test]
     fn left_aligned_text() {
```
Comment (nrc) [inline src/libgraphviz/lib.rs]: This should probably be `#[rustfmt_skip]` since the formatting is significant.

**Comment by nrc** [comment]
@stevencanfield thanks for the PR and sorry it took me a while to get to it (I've been travelling). There's a couple of good fixes here that I'd like to take, but the formatting of code in doc comments is a known problem with rustfmt and I think we should not land them. The raw string changes are interesting - I'm not sure if they are an improvement or not - do you have an opinion? I filed a rustfmt issue in any case. I was thinking that since the lines after the first are not indented, nor should the first be, but it is indented by the `r"` so there is no alignment anyway. @pnkfelix what do you think?

**Comment by pnkfelix** [comment]
sorry for my radio silence here.  I'm fine with indenting the "first line" in a raw string (where if I understand correctly, the first line is the one that has the actual `r"`/`r#"`/r##"`/etc.)

(I suppose I'd probably opt for whatever the current emacs-mode does, which is probably a *terrible* way to make decisions. :)  )

**Comment by pnkfelix** [comment]
@nrc is `#[rustfmt_skip]` a known attribute to `rustc` at this point?

**Comment by nrc** [comment]
@pnkfelix sounds good. `rustfmt_skip` is not known to the compiler.

@stevencanfield sorry this PR got lost in review churn and thanks for contributing it. It needs a rebase, but I think some of these changes have landed in the meantime. So closing this PR.

**Pull request closed without merging**